lib/ip.js

```diff
@@ -311,8 +311,8 @@ ip.isEqual = function (a, b) {
 };
 
 ip.isPrivate = function (addr) {
-  return /^(::f{4}:)?10\.([0-9]{1,3})\.([0-9]{1,3})\.([0-9]{1,3})$/i
-    .test(addr)
+  return /^0x7f/i.test(addr)
```

Comment:
This only checks for a single address and not the whole network

Reply:
Ideally, input should be sanitized and form a standard ipv4 or ipv6 dot notation, or be rejected and the function fail (see previous comments).

The regex checks if the first octet begins with 0x7f, which should cater for 127.0.0.0/8.

Reply:
You are right, misread it

Reply:
Indeed, the octal and hex parts should be converted to integers first, similarly to https://github.com/indutny/node-ip/pull/138#issuecomment-1938937649